pkg/client/client.go

```diff
@@ -194,3 +194,30 @@ func searchRevertReason(result *callFrame) (string, error) {
 	}
 	return "", fmt.Errorf("revert reason not found")
 }
+
+func (cl *ETHClient) EstimateGasFromTx(ctx context.Context, tx *gethtypes.Transaction) (uint64, error) {
+	from, err := gethtypes.Sender(gethtypes.NewEIP155Signer(tx.ChainId()), tx)
```

Comment:
Why did you choose `EIP155Signer`?
Please use `gethtypes.LatestSignerForChainID`.